### PR TITLE
Fake PR to use to comment on all code

### PR DESCRIPTION
**Pull request by neuhausneuhaus**

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// module.exports = {
```
Comment (neuhausneuhaus) [inline .eslintrc.js:1]: Looks like they were using eslint to clean and standardize their code by uncommenting this and having vscode lint the repo before making a commit.

Might be a good idea for us to do the same.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,27 @@
...
+
+~~SpyPress is a tool that can be run alongside any project to spy on requests and responses coming to and from the server. Using SpyPress is as easy as installing the npm package and requiring the the library to be used in your server file. SpyPress comes with a set of methods that allow the user to "listen" to the communications made with/by the server on the backend. Once incorporated into your project, all you need to do is run your server in development mode. Skypress will open up it’s own server on a localhost port to display all requests and responses via the front end in the browser~~
+
+### Who is SpyPress for?
+Spypress is meant for developers working with lightweight server-side frameworks. Developers can utilize SpyPress to visualize the servers they are creating and managing allowing them to optimize their servers through constant, real time monitoring of data passed from the client to server to database and back! 
+
+<strong> At present, visualization is limited to client requests.</strong>
+
+## What's next?
+- Implement a database to store each session in history for review at another time.
```
Comment (neuhausneuhaus) [inline README.md:25]: When put this way, i guess having a DB *sort of* makes sense. Let's call it a stretch stretch feature?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,27 @@
...
+- Improve websocket connections 
+- Visualize server responses.
```
Comment (neuhausneuhaus) [inline README.md:27]: 🙋
ON IT.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,27 @@
...
+~~SpyPress is a tool that can be run alongside any project to spy on requests and responses coming to and from the server. Using SpyPress is as easy as installing the npm package and requiring the the library to be used in your server file. SpyPress comes with a set of methods that allow the user to "listen" to the communications made with/by the server on the backend. Once incorporated into your project, all you need to do is run your server in development mode. Skypress will open up it’s own server on a localhost port to display all requests and responses via the front end in the browser~~
+
+### Who is SpyPress for?
+Spypress is meant for developers working with lightweight server-side frameworks. Developers can utilize SpyPress to visualize the servers they are creating and managing allowing them to optimize their servers through constant, real time monitoring of data passed from the client to server to database and back! 
+
+<strong> At present, visualization is limited to client requests.</strong>
+
+## What's next?
+- Implement a database to store each session in history for review at another time.
+- Improve websocket connections
```
Comment (neuhausneuhaus) [inline README.md:26]: I wonder what they meant by this. 🤷‍♀ Let's find out

File: client/styles.scss [hunk 5]
```diff
@@ -0,0 +1,73 @@
+@import url('https://fonts.googleapis.com/css?family=Ubuntu&display=swap');
```
Comment (osullid7) [inline client/styles.scss]: comments could be added for entire css file. nothing crazy, just to verbalise where each style is being referenced in the rest of our codebase.

File: client/index.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import App from "./App.jsx";
+import './styles.scss'
+
+//Render for main App Component, that has store inside as an attibute inside it's Provider;
+//Provides store to component;
+ReactDOM.render(
+  <Provider store={store}>
+    <App />
+  </Provider>,
+  document.querySelector("#root")
```
Comment (neuhausneuhaus) [inline client/index.js:22]: What is this `querySelector` for??

File: package.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    "redux-thunk": "^2.3.0",
+    "sass-loader": "^7.3.1",
+    "style-loader": "^1.0.0",
+    "webpack": "^4.39.2",
+    "webpack-cli": "^3.3.7",
+    "webpack-dev-server": "^3.8.0",
+    "webpack-hot-middleware": "^2.25.0"
+  },
+  "bugs": {
+    "url": "https://github.com/Spypress/Spypress/issues"
```
Comment (neuhausneuhaus) [inline package.json:71]: Never seen this before. Interesting.
I'm guessing that this adds a link to the sidebar of their NPM page?

File: client/index.js (hunk 6 above)
Comment (osullid7) [reply client/index.js:22]: i think this is how the app is being rendered to our html file, linked through the id 'root'. used to seeing it as getElementById i believe, but phil would take marks off for that.

File: public/socket.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+//Middle WebSocket connection for server and client;
+
+//Establish new server on same port as server;
+const socket = new WebSocket("ws://localhost:2000/");
+
+//Listening for when server is opened
+socket.addEventListener('open', (event) => {
+  socket.send('Hello Server');
+});
```
Comment (neuhausneuhaus) [inline public/socket.js:9]: @sam-parsons I guess this is what you do when you establish a two way connection when you only really needed one? I wonder if everything would work without this. 
If so, wouldn't that effectively be the "server sent events" setup you were looking for?

File: server/server.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+const wss = new WebSocket.Server({ port: 2000 });
+
+//Event listener for WebSocket Connection.
+wss.on("connection", function connection(ws) { 
+
+  //ON Connection parse all requests & cookies
+  app.use(express.json());
+  app.use(cookieParser());
+  //Listen on all Requests
+  app.use("*", (req, res, next) => {
```
Comment (neuhausneuhaus) [inline server/server.js]: I feel like this _should_ be extracted to a middleware function elsewhere, but I'm okay with it at least until there are other middleware functions

File: server/server.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+    //All set as null until we define the data coming in.
+    const data = {
+      header: null,
+      cookies: null,
+      body: null,
+      type:null
+    };
+
+    //Defining any requests that comes in such as Methods, Headers, Cookies & Body.
+    !req.method ? data.type = 'Method not sent' : data.type = req.method;
```
Comment (neuhausneuhaus) [inline server/server.js:57]: these can be rewritten more succinctly as
```javascript
    data.type = !req.method ? 'Method not sent' : req.method;

```

Comment (neuhausneuhaus) [reply server/server.js:57]: But also, who cares. This code is immaculate you guys...

File: client/index.js (hunk 6 above)
Comment (neuhausneuhaus) [reply client/index.js:22]: Oh I see.

File: client/styles.scss (hunk 5 above)
Comment (neuhausneuhaus) [reply client/styles.scss]: I mean, I guess? The way I see it, the fact that there are class name selectors are comment enough.

**Comment by neuhausneuhaus** [comment]
😍 This is good stuff!

File: client/containers/Connection.jsx [hunk 11]
```diff
@@ -0,0 +1,57 @@
+import React, { Component } from "react";
+import { connect } from "react-redux"; //Will need to use mapStateToProps & mapDispatchToProps
```
Comment (osullid7) [inline client/containers/Connection.jsx:2]: Connection.jsx has access to state through the store, which is prop drilled down on line 44 to the Message component, giving Message access to state

File: client/actions/actions.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+ * ************************************
+ */
+
+//Import all types from constant file (actionTypes);
+import * as types from "./actionTypes";
+
+//For reducer to grab and use;
+export const addMessage = message => ({
+  type: types.ADD_MESSAGE,
+  payload: message // Data OBJECT we sent from the server;
```
Comment (osullid7) [inline client/actions/actions.js:16]: how does this object get received from the server?

**Comment by neuhausneuhaus** [comment]
Closing

**Pull request closed without merging**